cmd/event-logs/main.go

```diff
@@ -0,0 +1,174 @@
+)
+
+const (
+	svcName        = "event-logs"
+	envPrefixDB    = "MG_EVENT_LOGS_"
+	envPrefixHTTP  = "MG_EVENT_LOGS_HTTP_"
+	envPrefixAuth  = "MG_AUTH_GRPC_"
+	defDB          = "events"
+	defSvcHTTPPort = "9021"
+)
+
+type config struct {
+	LogLevel      string  `env:"MG_EVENT_LOGS_LOG_LEVEL"     envDefault:"info"`
+	ESURL         string  `env:"MG_ES_URL"                   envDefault:"nats://localhost:4222"`
+	JaegerURL     url.URL `env:"MG_JAEGER_URL"               envDefault:"http://jaeger:14268/api/traces"`
+	SendTelemetry bool    `env:"MG_SEND_TELEMETRY"           envDefault:"true"`
+	InstanceID    string  `env:"MG_EVENT_LOGS_INSTANCE_ID"   envDefault:""`
+	TraceRatio    float64 `env:"MG_JAEGER_TRACE_RATIO"       envDefault:"1.0"`
+}
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	g, ctx := errgroup.WithContext(ctx)
+
+	cfg := config{}
+	if err := env.Parse(&cfg); err != nil {
+		log.Fatalf("failed to load %s configuration : %s", svcName, err)
+	}
+
+	logger, err := mglog.New(os.Stdout, cfg.LogLevel)
+	if err != nil {
+		log.Fatalf("failed to init logger: %s", err)
+	}
+
+	var exitCode int
+	defer mglog.ExitWithError(&exitCode)
+
+	if cfg.InstanceID == "" {
+		if cfg.InstanceID, err = uuid.New().ID(); err != nil {
+			logger.Error("failed to generate instanceID: %s", err)
+			exitCode = 1
+			return
+		}
+	}
+
+	dbConfig := pgclient.Config{Name: defDB}
+	if err := env.ParseWithOptions(&dbConfig, env.Options{Prefix: envPrefixDB}); err != nil {
+		logger.Error("failed to load %s Postgres configuration : %s", svcName, err)
+		exitCode = 1
+		return
+	}
+	db, err := pgclient.Setup(dbConfig, *eventlogspg.Migration())
+	if err != nil {
+		logger.Error(err.Error())
+		exitCode = 1
+		return
+	}
+	defer db.Close()
+	repo := eventlogspg.NewRepository(db)
+
+	authConfig := auth.Config{}
+	if err := env.ParseWithOptions(&authConfig, env.Options{Prefix: envPrefixAuth}); err != nil {
+		logger.Error("failed to load %s auth configuration : %s", svcName, err)
+		exitCode = 1
+		return
+	}
+
+	ac, acHandler, err := auth.Setup(authConfig)
+	if err != nil {
+		logger.Error(err.Error())
+		exitCode = 1
+		return
+	}
+	defer acHandler.Close()
+
+	logger.Info("Successfully connected to auth grpc server " + acHandler.Secure())
+
+	tp, err := jaegerclient.NewProvider(ctx, svcName, cfg.JaegerURL, cfg.InstanceID, cfg.TraceRatio)
+	if err != nil {
+		logger.Error("Failed to init Jaeger: %s", err)
+		exitCode = 1
+		return
+	}
+	defer func() {
+		if err := tp.Shutdown(ctx); err != nil {
+			logger.Error("Error shutting down tracer provider: %v", err)
+		}
+	}()
+	tracer := tp.Tracer(svcName)
+
+	svc := newService(repo, ac, logger, tracer)
+
+	subscriber, err := store.NewSubscriber(ctx, cfg.ESURL, logger)
+	if err != nil {
+		logger.Error("failed to create subscriber: %s", err)
+		exitCode = 1
+		return
+	}
+
+	logger.Info("Subscribed to Event Store")
+
+	if err := eventlogs.Start(ctx, svcName, subscriber, repo); err != nil {
```

Comment:
Add event processing to service, rather than using `repo` directly. Also, add logging to it.